public/js/register.js

```diff
@@ -0,0 +1,106 @@
+$(document).ready(function() {
+    registerInputListener('email', validateEmail, 'emaildiv', getEmailError);
+    registerInputListener('username', validateUsername, 'usernamediv', getUsernameError);
+    registerInputListener('password',  validatePassword, 'passworddiv', getPasswordError);
+    registerInputListener('password', validatePasswordRepeat, 'password_repeatdiv', getPasswordRepeatError);
+    registerInputListener('password_repeat', validatePasswordRepeat, 'password_repeatdiv', getPasswordRepeatError);
+});
+
+function registerInputListener(elementid, validator, divid, error) {
+  const listener = createInputListener(validator, divid, error);
+  const element = $('#' + elementid);
+  element.change(listener);
+  element.keyup(listener);
+}
+
+function createInputListener(validator, divid, error) {
+  return function () {
+    decorateErrorDiv(validator(), divid, error());
+    validate();
+  };
+}
+
+function validate() {
+  const validation = (validateEmail() & validateUsername() & validatePassword() & validatePasswordRepeat());
```

Comment:
Sollten hier nicht boolsche AND Operationen genutzt werden anstatt Bitwise AND ?